### PR TITLE
Added claims, and various shared enums

### PR DESCRIPTION
**Pull request by dan-stone**

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -1,20 +1,30 @@
...
+        assert.typeOf(module, 'object');
+
+        // Enums
+        assert.typeOf(module.Roles, 'object');
+        assert.typeOf(module.Scopes, 'object');
+        assert.typeOf(module.ResourceActions, 'object');
 
         // Middleware
-        assert.typeOf(index.createMiddleware, 'function');
+        assert.typeOf(module.createMiddleware, 'function');
```
Comment (dan-stone) [inline test/index.test.js]: should the `createMiddleware` function perhaps be renamed to `createJwtMiddleware`?

Comment (MikeVezina) [reply test/index.test.js]: Yes! Great suggestion

File: lib/claims/index.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+
+function IdentityClaim({user}) {
+    this.user = user
+}
+IdentityClaim.prototype.assertPermission = function({resourceOwner, scope, action=ResourceActions.READ}) {
+    if (resourceOwner === this.user.id) {
+        return
+    }
+    if (this.user.roles.includes('ADMIN')) {
```
Comment (MikeVezina) [inline lib/claims/index.js]: Should utilize Roles.ADMIN, rather than 'ADMIN'

File: lib/claims/index.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        return
+    }
+    if (this.user.roles.includes('DATA_ADMIN') && action === ResourceActions.READ) {
```
Comment (MikeVezina) [inline lib/claims/index.js]: Should utilize Roles.DATA_ADMIN, rather than 'DATA_ADMIN'

File: lib/claims/index.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    throw new NotAuthorized(``)
+}
+
+
+function GrantClaim({owner, client, scopes}) {
+    this.owner = owner
+    this.client = client
+    this.scopes = scopes
+}
+GrantClaim.prototype.assertPermission = function({resourceOwner, scope, action=ResourceActions.READ}) {
```
Comment (MikeVezina) [inline lib/claims/index.js]: 'action' parameter is unused. Is it necessary for other reasons?

File: lib/claims/index.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    if (this.user.roles.includes(Roles.DATA_ADMIN) && action === ResourceActions.READ) {
+        return
+    }
+    throw new NotAuthorized(``)
+}
+
+
+function GrantClaim({owner, client, scopes}) {
+    this.owner = owner
+    this.client = client
```
Comment (MikeVezina) [inline lib/claims/index.js]: Is `this.client` supposed to be used somewhere?

File: lib/claims/index.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        throw new NotAuthorized(``)
+    }
+    if (!this.scopes.includes(scope)) {
+        throw new NotAuthorized(``)
+    }
+}
+
+
+function GroupClaim({user, group}) {
+    this.user = user
```
Comment (MikeVezina) [inline lib/claims/index.js]: `this.user` is also not used

File: lib/claims/index.js (hunk 5 above)
Comment (dan-stone) [reply lib/claims/index.js]: No, it's just part of the claim JSON structure.  We'll want to track usage by 3rd party clients in the long run (for billing, rate limiting, analytics, etc), so this needs to be exposed to services, even though it's not a part of the permissions

File: lib/claims/index.js (hunk 6 above)
Comment (dan-stone) [reply lib/claims/index.js]: I added it for robustness (for example, logging purposes by services)

**Pull request merged**